### PR TITLE
rusk-wallet: Fix creation of deploy-transaction

### PR DESCRIPTION
**Pull request by moCello**
This pick up the changes to the deploy-transaction creation from #2540 after it has been reverted and separated from the changes made to the syncing (#2558)

### CODE REVIEW

File: rusk-wallet/src/bin/command.rs [hunk 1]
```diff
@@ -944,6 +996,9 @@ impl fmt::Display for RunResult {
                 let reward = Dusk::from(data.reward);
                 write!(f, "> Accumulated reward is: {reward} DUSK")
             }
+            ContractId(bytes) => {
+                write!(f, "> Contract ID: {:?}", bytes)
```
Comment (miloszm) [inline rusk-wallet/src/bin/command.rs:1000]: this needs to be displayed as 64-character hexadecimal (currently it is an array if decimal bytes)

File: rusk-wallet/src/bin/interactive.rs [hunk 2]
```diff
@@ -707,6 +711,20 @@ fn confirm(cmd: &Command) -> anyhow::Result<bool> {
             println!("   > Max fee = {} DUSK", Dusk::from(max_fee));
             prompt::ask_confirm()
         }
+        Command::MoonlightStake {
```
Comment (miloszm) [inline rusk-wallet/src/bin/interactive.rs:714]: this command does not belong in this PR, it has nothing to do with the deployment

File: rusk-wallet/src/bin/interactive.rs [hunk 3]
```diff
@@ -718,6 +736,18 @@ fn confirm(cmd: &Command) -> anyhow::Result<bool> {
             println!("   > Max fee = {} DUSK", Dusk::from(max_fee));
             prompt::ask_confirm()
         }
+        Command::MoonlightUnstake {
```
Comment (miloszm) [inline rusk-wallet/src/bin/interactive.rs:739]: same as above, and several others, I think these commands should be added in a relevant PR

File: rusk-wallet/src/bin/command.rs [hunk 4]
```diff
@@ -226,7 +231,7 @@ pub(crate) enum Command {
         fn_args: Vec<u8>,
 
         /// Max amount of gas for this transaction
-        #[clap(short = 'l', long, default_value_t = DEFAULT_STAKE_GAS_LIMIT)]
+        #[clap(short = 'l', long, default_value_t = DEFAULT_LIMIT_STAKE)]
```
Comment (miloszm) [inline rusk-wallet/src/bin/command.rs:234]: why is stake-related constant used in an object which is not stake-related

File: rusk-wallet/src/bin/command.rs [hunk 5]
```diff
@@ -410,14 +419,30 @@ pub(crate) enum Command {
         amt: Dusk,
 
         /// Max amount of gas for this transaction
-        #[clap(short = 'l', long, default_value_t = DEFAULT_STAKE_GAS_LIMIT)]
+        #[clap(short = 'l', long, default_value_t = DEFAULT_LIMIT_STAKE)]
```
Comment (miloszm) [inline rusk-wallet/src/bin/command.rs:422]: as above, stake-related constant should not be used in object not related to staking

File: rusk-wallet/src/bin/command.rs [hunk 6]
```diff
@@ -429,7 +454,7 @@ pub(crate) enum Command {
...
```
Comment (miloszm) [inline rusk-wallet/src/bin/command.rs:457]: as above, stake-related constant shouldn't be used here

**Review by miloszm — CHANGES_REQUESTED** [review]

**Comment by miloszm** [comment]
I do not see the actual deployment implemented in this PR, it only contains a way to generate contract id. Is there another PR which contains the deployment proper?

File: rusk-wallet/src/bin/command.rs (hunk 1 above)
Comment (Daksh14) [reply rusk-wallet/src/bin/command.rs]: @miloszm Using hex::encode() now

File: rusk-wallet/src/bin/interactive.rs (hunk 2 above)
Comment (Daksh14) [reply rusk-wallet/src/bin/interactive.rs]: @miloszm Removed

File: rusk-wallet/src/bin/interactive.rs (hunk 3 above)
Comment (Daksh14) [reply rusk-wallet/src/bin/interactive.rs]: @miloszm Removed

File: rusk-wallet/src/bin/command.rs (hunk 4 above)
Comment (Daksh14) [reply rusk-wallet/src/bin/command.rs]: @miloszm Changed to `DEFAULT_LIMIT`

File: rusk-wallet/src/bin/command.rs (hunk 5 above)
Comment (Daksh14) [reply rusk-wallet/src/bin/command.rs]: same

File: rusk-wallet/src/bin/command.rs (hunk 6 above)
Comment (Daksh14) [reply rusk-wallet/src/bin/command.rs]: same

**Review by miloszm — APPROVED** [review]
Nice, I was able to deploy a contract via Phoenix, 1 remark:
- why isn't the wallet checking the status of sent transaction? - it could tell the user that transaction was successful or if not, what was the error
- why does note fetching take so long? I'd compare with how it is done in deployment tool as it seems to be much faster there
- status feedback during deployment is flashing messages too quickly, it is not readable, and it is showing msg about streaming when it is actually doing proving, it'd be better to show more static messages about what the wallet is actually doing

**Comment by moCello** [comment]
>     * why isn't the wallet checking the status of sent transaction? - it could tell the user that transaction was successful or if not , was is the error

@HDauven also pointed to me to this. He was going to add an issue

File: rusk-wallet/src/wallet/gas.rs [hunk 7]
```diff
@@ -10,10 +10,16 @@
...
 
 /// The default gas limit
-pub const DEFAULT_LIMIT: u64 = 500_000_000;
+pub const DEFAULT_LIMIT: u64 = 100_000_000;
+
+/// The default gas limit for a contract deployment
+pub const DEFAULT_LIMIT_DEPLOYMENT: u64 = 50_000_000;
+
+/// The default gas limit for staking
+pub const DEFAULT_LIMIT_STAKE: u64 = 2_900_000_000;
```
Comment (moCello) [inline rusk-wallet/src/wallet/gas.rs:22]: Looking at the PR a bit closer, this limit is not only being used for staking operations but also other contract-calls. The variable should hence be called something like `DEFAULT_LIMIT_CONTRACT_CALL` (or shorter `DEFAULT_LIMIT_CALL`)

File: rusk-wallet/src/wallet/gas.rs [hunk 8]
```diff
@@ -10,10 +10,16 @@
 use crate::currency::Lux;
 
 /// The minimum gas limit
-pub const MIN_LIMIT: u64 = 350_000_000;
+pub const MIN_LIMIT: u64 = 50_000_000;
 
 /// The default gas limit
-pub const DEFAULT_LIMIT: u64 = 500_000_000;
+pub const DEFAULT_LIMIT: u64 = 100_000_000;
```
Comment (moCello) [inline rusk-wallet/src/wallet/gas.rs:16]: with the addition of other default limits, it might make sense to call this one `DEFAULT_LIMIT_TRANSACTION` and adjust the doc accordingly

**Comment by HDauven** [comment]
> > ```
> > * why isn't the wallet checking the status of sent transaction? - it could tell the user that transaction was successful or if not , was is the error
> > ```
> 
> @HDauven also pointed to me to this. He was going to add an issue

Sorry @miloszm @moCello, had to participate in a long meeting before I was able to finalize the issue. Here it is: https://github.com/dusk-network/rusk/issues/2600

File: rusk-wallet/src/wallet/gas.rs (hunk 8 above)
Comment (Daksh14) [reply rusk-wallet/src/wallet/gas.rs:16]: Created issue https://github.com/dusk-network/rusk/issues/2603

File: rusk-wallet/src/bin/command.rs (hunk 4 above)
Comment (moCello) [reply rusk-wallet/src/bin/command.rs]: I don't think it is a good idea to take the default gas limit here, since the default limit is the default limit for single transaction. Any contract call (like for example any stake operation) will consume more gas points which is why the default stake limit makes sense here but would need a different name (like I mentioned in my other review comments)

File: rusk-wallet/src/bin/command.rs (hunk 5 above)
Comment (moCello) [reply rusk-wallet/src/bin/command.rs]: same

File: rusk-wallet/src/bin/command.rs (hunk 6 above)
Comment (moCello) [reply rusk-wallet/src/bin/command.rs]: and here too

File: rusk-wallet/src/bin/command.rs [hunk 9]
```diff
@@ -410,14 +419,30 @@ pub(crate) enum Command {
         amt: Dusk,
 
         /// Max amount of gas for this transaction
-        #[clap(short = 'l', long, default_value_t = DEFAULT_STAKE_GAS_LIMIT)]
+        #[clap(short = 'l', long, default_value_t = DEFAULT_LIMIT)]
```
Comment (moCello) [inline rusk-wallet/src/bin/command.rs:422]: Here we most likely also need more gas than in the default limit

File: rusk-wallet/src/bin/command.rs [hunk 10]
```diff
@@ -429,7 +454,7 @@ pub(crate) enum Command {
...
```
Comment (moCello) [inline rusk-wallet/src/bin/command.rs:457]: same

**Review by moCello — COMMENTED** [review]
I am still not happy with the const names for the default gas values.
The const for gas-price now named `DEFAULT_PRICE_TRANSACTION` suggests that this value should be used everywhere where we also use `DEFAULT_LIMIT_TRANSACTION` and wise versa. But the value for `DEFAULT_LIMIT_TRANSACTION` is only used in transfer-transactions whereas the `DEFAULT_PRICE_TRANSACTION` is used for *all* transactions.
The context in which these two constants are used is different and their name needs to reflect that.

File: rusk-wallet/src/bin/command.rs [hunk 11]
```diff
@@ -410,14 +420,30 @@ pub(crate) enum Command {
         amt: Dusk,
 
         /// Max amount of gas for this transaction
-        #[clap(short = 'l', long, default_value_t = DEFAULT_STAKE_GAS_LIMIT)]
+        #[clap(short = 'l', long, default_value_t = DEFAULT_LIMIT_TRANSFER)]
```
Comment (moCello) [inline rusk-wallet/src/bin/command.rs:423]: Conversions are managed by contract-calls too, so the `DEFAULT_LIMIT_CALL` should be used here

File: rusk-wallet/src/bin/command.rs [hunk 12]
```diff
@@ -429,7 +455,7 @@ pub(crate) enum Command {
...
```
Comment (moCello) [inline rusk-wallet/src/bin/command.rs:458]: Same as with `PhoenixToMoonlight`

File: rusk-wallet/src/bin/interactive.rs [hunk 13]
```diff
@@ -489,18 +498,29 @@ fn menu_op(
             AddrOp::Run(Box::new(Command::MoonlightToPhoenix {
                 addr_idx: Some(addr_idx),
                 amt: prompt::request_token_amt("convert", moonlight_balance)?,
-                gas_limit: prompt::request_gas_limit(gas::DEFAULT_LIMIT)?,
+                gas_limit: prompt::request_gas_limit(
+                    gas::DEFAULT_LIMIT_TRANSFER,
```
Comment (moCello) [inline rusk-wallet/src/bin/interactive.rs:502]: here too

File: rusk-wallet/src/bin/interactive.rs [hunk 14]
```diff
@@ -489,18 +498,29 @@ fn menu_op(
...
                 gas_price: prompt::request_gas_price()?,
             }))
         }
         CMI::PhoenixToMoonlight => {
             AddrOp::Run(Box::new(Command::PhoenixToMoonlight {
                 addr_idx: Some(addr_idx),
                 amt: prompt::request_token_amt("convert", phoenix_balance)?,
-                gas_limit: prompt::request_gas_limit(gas::DEFAULT_LIMIT)?,
+                gas_limit: prompt::request_gas_limit(
+                    gas::DEFAULT_LIMIT_TRANSFER,
```
Comment (moCello) [inline rusk-wallet/src/bin/interactive.rs:512]: and here too

File: rusk-wallet/src/wallet.rs [hunk 15]
```diff
@@ -977,8 +979,15 @@ impl<F: SecureWalletFile + Debug> Wallet<F> {
...
         let deploy = moonlight_deployment(
-            &sender_sk, bytes_code, pk, init_args, gas.limit, gas.price, nonce,
-            0, chain_id,
+            &sender_sk,
+            bytes_code,
+            pk,
+            init_args,
+            gas.limit,
+            gas.price,
+            nonce,
```
Comment (moCello) [inline rusk-wallet/src/wallet.rs:988]: now that there are two nonces this one should be renamed `moonlight_nonce`

File: rusk-wallet/src/wallet/gas.rs (hunk 8 above)
Comment (moCello) [reply rusk-wallet/src/wallet/gas.rs]: closed issue #2603 again since these changes will be handled in this PR

**Review by Daksh14 — APPROVED** [review]
LGTM

**Review by HDauven — APPROVED** [review]
LGTM
